Moved block pruner fetching outside the try with resources of an update

rolfyone opened the pull request:
This should mean the fetch query is no longer blocking an update.

Also extended frequency to 4-hourly, which is around 1200 blocks.

I had considered removing the need for the index scan by utilising the block column, but this may be worse, as it has to compute the block root for every block returned... In the first cleanup instance it would likely mean a large amount of computation.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

rolfyone committed:
Moved block pruner fetching outside the try with resources of an update

This should mean the fetch query is no longer blocking an update.

Also extended frequency to 4-hourly, which is around 1200 blocks.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
removed test case that's no longer needed.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

lucassaldanha reviewed (commented):
The change looks ok to me. However, we deleted a test for the functionality. Do we have other tests to ensure we are not breaking the desired behaviour?

rolfyone committed:
Merge branch 'master' into pruning-refactor

StefanBratanov committed:
Remove hidden option `--Xbeacon-liveness-tracking-enabled` (#6926)

lucassaldanha reviewed (approved):
LGTM

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into pruning-refactor

rolfyone committed:
implement an alternative pruning strategy for smaller block sets to prune. This will require more cpu to read block, but less of a scan, so for small numbers of blocks should be quite clean.

Also adjusted the pruning frequency to keep the numbers small, which is what will make this method practical.

In 15 minutes, should be less than 75 blocks to prune, so the 500 limit for pruning by this new method should easily handle the small number of blocks that trickle in.

Larger numbers can still benefit from using the index, but for smaller scans it's a large overhead, because of the need to scan the entire index.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Earliest block not always getting fetched, add debug and let it use prune by stream if earliest block not returned

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Reorganise pruning loops

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
fix debug

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into pruning-refactor
